process/smartContract/process.go

```diff
@@ -1592,9 +1597,34 @@ func (sc *scProcessor) processVMOutput(
 		return nil, err
 	}
 
+	err = sc.checkSCRSizeInvariant(scrTxs)
+	if err != nil {
+		return nil, err
+	}
+
 	return scrTxs, nil
 }
 
+func (sc *scProcessor) checkSCRSizeInvariant(scrTxs []data.TransactionHandler) error {
+	if !sc.flagSCRSizeInvariantCheck.IsSet() {
+		return nil
+	}
+
+	for _, scrHandler := range scrTxs {
+		scr, ok := scrHandler.(*smartContractResult.SmartContractResult)
+		if !ok {
+			return process.ErrWrongTypeAssertion
+		}
+
+		lenTotalData := len(scr.Data) + len(scr.ReturnMessage) + len(scr.Code)
+		if lenTotalData > core.MegabyteSize {
```

Comment:
1MB is too much as there are other fields + other parts of the message that need to get marshaled.
Suggestion is to have another stand-alone const in common, something like
```GO
 const MaxTotalSCRsSize = 3 * (1 << 18) //768KB
```

Reply:
done